proving_system/stark/src/cairo_run/parser.rs

```diff
@@ -0,0 +1,187 @@
+use super::file_writer::FileWriter;
+use cairo_vm::cairo_run::{self, EncodeTraceError};
+use cairo_vm::hint_processor::builtin_hint_processor::builtin_hint_processor_definition::BuiltinHintProcessor;
+use cairo_vm::vm::errors::cairo_run_errors::CairoRunError;
+use cairo_vm::vm::errors::trace_errors::TraceError;
+use cairo_vm::vm::errors::vm_errors::VirtualMachineError;
+use std::io;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Failed to interact with the file system")]
+    IO(#[from] std::io::Error),
+    #[error("The cairo program execution failed")]
+    Runner(#[from] CairoRunError),
+    #[error(transparent)]
+    EncodeTrace(#[from] EncodeTraceError),
+    #[error(transparent)]
+    VirtualMachine(#[from] VirtualMachineError),
+    #[error(transparent)]
+    Trace(#[from] TraceError),
+}
+
+pub fn parse_cairo_file(
```

Comment:
Rather than parsing, a better name could be something like `run_program`, and a little documentation explaining that runs a compiled cairo program with cairo-rs and it writes two files into disk, one for the memory and one for the trace would be nice